bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/PowerControlPower.java

```diff
@@ -69,13 +68,42 @@ public void onReceiveCommand(String channelId, LGWebOSHandler handler, Command c
                         }, i * WOL_PACKET_RETRY_DELAY_MILLIS, TimeUnit.MILLISECONDS);
                     }
                 }
+            } else {
+                switch (state) {
+                    case CONNECTING:
+                    case REGISTERING:
+                    case REGISTERED:
+                        logger.debug("Received ON - TV is already on.");
+                        handler.postUpdate(channelId, OnOffType.ON);
+                        break;
+                    case DISCONNECTING:
+                        logger.debug("Received ON - TV is currently disconnecting.");
+                        handler.postUpdate(channelId, OnOffType.OFF);
+                        break;
+                    case DISCONNECTED:
+                        // none reachable code
+                        break;
+                }
             }
         } else if (OnOffType.OFF == command) {
-            if (!handler.getSocket().isConnected()) {
-                logger.debug("Received OFF - TV is already off.");
-                handler.postUpdate(channelId, OnOffType.OFF);
-            } else {
+            if (State.REGISTERED == state || State.REGISTERING == state) {
                 handler.getSocket().powerOff(getDefaultResponseListener());
+            } else {
+                switch (state) {
+                    case CONNECTING:
+                        logger.debug("Received OFF - TV is currently connecting.");
+                        handler.postUpdate(channelId, OnOffType.ON);
```

Comment:
No coherent with your logic.

Reply:
you proposed:
```
            if (!handler.getSocket().isConnected()) {
                logger.debug("Received OFF - TV is already off.");
                handler.postUpdate(channelId, OnOffType.OFF);
            } else {
                handler.getSocket().powerOff(getDefaultResponseListener());
            }
```
powerOff would be executed in states CONNECTING, REGISTERING, REGISTERED and DISCONNECTING but it would only actually send a message in REGISTERED state.

in case DISCONNECTED it would not attempt to send sth and update the channel to OFF. 
Indeed this should work as well.

so the difference now is instead of no operation in states
CONNECTING, REGISTERING, DISCONNECTING

CONNECTING, REGISTERING would update the state to ON <- we definitely should not set it to OFF
and DISCONNECTING to OFF <- same as in onStateChanged

This should not cause any issue, but happy to modify the code and not update the channel in those cases

Reply:
just thought it should be updated in analogy to the ON command case

Reply:
Your logic since the beginning is that the power channel must be set to ON only when the state is REGISTERED, that means when the binding is ready to handle commands. In all other cases, it must be OFF. I finally agree with your initial logic.

So if you receive a command on this channel when the current state is other than REGISTERED, you must reject the command and you must update the channel to OFF.